server/src/server.rs

```diff
@@ -166,27 +168,35 @@ async fn handle_stream(
         {
             use crate::cloud::do_cloud_auth;
             use log::debug;
+            use selium_protocol::error_codes::CLOUD_AUTH_FAILED;
 
             match do_cloud_auth(&_connection, topic, &topics).await {
                 Ok(_) => stream.send(Frame::Ok).await?,
                 Err(e) => {
                     debug!("Cloud authentication error: {e:?}");
 
-                    stream
-                        .send(Frame::Error(e.to_string().into_bytes().into()))
-                        .await?;
+                    let payload = ErrorPayload {
+                        code: CLOUD_AUTH_FAILED,
+                        message: e.to_string().into(),
+                    };
+
+                    stream.send(Frame::Error(payload)).await?;
 
                     return Ok(());
                 }
             }
         }
         #[cfg(not(feature = "__cloud"))]
         {
+            use selium_protocol::error_codes::REPLIER_ALREADY_BOUND;
+
             // Note this can only occur if someone circumvents the client lib
             if !topic.is_valid() {
-                stream
-                    .send(Frame::Error(b"Invalid topic name".to_vec().into()))
-                    .await?;
+                let payload = ErrorPayload {
+                    code: REPLIER_ALREADY_BOUND,
```

Comment:
Is this the error code you want? Shouldn't it be `INVALID_TOPIC_NAME`?

Reply:
Yes, yes it should 🤦‍♂️

Reply:
Fixed!